### PR TITLE
Iv 1209 tss

### PR DESCRIPTION
**Pull request by psschroeter**

### CODE REVIEW

File: rll/collectd.sh [hunk 1]
```diff
@@ -259,11 +263,22 @@ else
   echo "swapfile not setup, skipping collectd swap plugin"
 fi
 
+# Populate RS_RLL_PORT
+source /var/run/rightlink/secret
+/usr/local/bin/rsc --rl10 rl10 put_hostname /rll/tss/hostname hostname=$COLLECTD_SERVER
```
Comment (Maimer) [inline rll/collectd.sh]: When using an rl10 call through RSC I believe it knows to use the RL10 creds already.  So you shouldn't need to source the secret file and you shouldn't need to pass the --rl10 flag.  The --rl10 flag is used to make calls to the cm15 (or other) APIs using the rl10 creds.

File: rll/setup_software_repo.sh [hunk 2]
```diff
@@ -1,62 +1,37 @@
...
-    curl -s http://mirror.rightscale.com/rightlink/rightscale.pub | sudo apt-key add -
-  else
-    wget -q -O- http://mirror.rightscale.com/rightlink/rightscale.pub | sudo apt-key add -
-  fi
-  sudo dd of=/etc/apt/sources.list.d/rightscale.sources.list <<EOF
-deb [arch=amd64] http://mirror.rightscale.com/rightscale_software_ubuntu/latest $distro_codename main
-deb-src [arch=amd64] http://mirror.rightscale.com/rightscale_software_ubuntu/latest $distro_codename main
-EOF
+  # Autopopulate this ENV var for all subsequent scripts
+  rsc --rl10 rl10 update /rll/env/DEBIAN_FRONTEND payload=noninteractive
```
Comment (Maimer) [inline rll/setup_software_repo.sh]: Same as above, I don't believe you need the --rl10 flag here.

**Comment by Maimer** [comment]
A general comment about using RSC in all these scripts: should we be using sudo to before each rsc call? (would apply to all the scripts in this repo)

File: rll/collectd.sh (hunk 1 above)
Comment (psschroeter) [inline rll/collectd.sh]: This is a little deceptive. the secret file isn't source for rsc, its sourced for writing the collectd config 3-4 lines below this.

**Comment by psschroeter** [comment]
Sudo shouldn't be needed, scripts run as the same user as the secrets file.

File: CHANGES.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 Changelog for the RightLink10 Base ServerTemplate
 =================================================
 
+10.1.1
```
Comment (lopaka) [inline CHANGES.md]: should be `10.1.rc1`

File: rll/collectd.sh [hunk 4]
```diff
@@ -259,11 +263,22 @@ else
   echo "swapfile not setup, skipping collectd swap plugin"
 fi
 
+# Populate RS_RLL_PORT
+source /var/run/rightlink/secret
```
Comment (lopaka) [inline rll/collectd.sh:309]: This will not work as the non-root user, correct?

File: rll/setup_software_repo.sh [hunk 5]
```diff
@@ -1,62 +1,37 @@
-#! /bin/bash
+#! /bin/bash -e
 
-# Add RightScale software repository, primarily used for installing monitoring (collectd).
-
-set -ex
+# Install security updates
+# Set up needed external repositories for collectd.
```
Comment (lopaka) [inline rll/setup_software_repo.sh]: Let's remove this comment since nothing needed for collectd after all.

**Comment by lopaka** [comment]
We should rename `rll/setup_software_repo.sh` to `rll/install_security_updates.sh`

File: rll/setup_software_repo.sh [hunk 6]
```diff
@@ -92,22 +60,7 @@ EOF
...
-    time sudo yum -y install yum-plugin-security
-    # update-minimal fails on RHEL7, see https://bugzilla.redhat.com/show_bug.cgi?id=1048584
-    time sudo yum -y --security update
-    ;;
-  esac
-
+  time sudo yum -y install yum-plugin-security
+  # update-minimal fails on RHEL7, see https://bugzilla.redhat.com/show_bug.cgi?id=1048584
+  time sudo yum -y --security update
 fi
```
Comment (lopaka) [inline rll/setup_software_repo.sh]: Can we have an `else` section logging that unable to determine OS or packaging software, no updates installed?

File: rll/collectd.sh (hunk 4 above)
Comment (psschroeter) [inline rll/collectd.sh:309]: Its owned by rightlink and the script is run as rightlink, so works in this case

Comment (lopaka) [inline rll/collectd.sh:309]: Perfect!

**Pull request merged**